e2e/pages/volumes-page.ts

```diff
@@ -0,0 +1,140 @@
+import type {Locator, Page} from '@playwright/test';
+
+export class VolumesPage {
+  readonly page: Page;
+  readonly table: Locator;
+  readonly namespaceSelector: Locator;
+  readonly searchBox: Locator;
+  readonly errorBanner: Locator;
+
+  constructor(page: Page) {
+    this.page = page;
+    this.table = page.locator('.volumesTable');
+    this.namespaceSelector = page.locator('.select-namespace');
+    this.searchBox = page.locator('.search-box input');
+    this.errorBanner = page.locator('.banner.error');
+  }
+
+  getVolumeRow(volumeName: string) {
+    return this.page.locator('tr.main-row').filter({hasText: volumeName});
+  }
+
+  async waitForVolumeToAppear(volumeName: string, timeout = 30000) {
+    const volumeRow = this.getVolumeRow(volumeName);
+    await volumeRow.waitFor({state: 'visible', timeout});
+  }
+
+  async clickVolumeAction(volumeName: string, action: string) {
+    const volumeRow = this.getVolumeRow(volumeName);
+    const actionButton = volumeRow.locator('.btn.role-multi-action');
+    await actionButton.click();
+
+    await this.page.waitForSelector('.list-unstyled', {state: 'visible', timeout: 2000});
+
+    const actionText = action === 'browse' ? 'Browse Files' :
+        action === 'delete' ? 'Delete' : action;
+
+    const actionLocator = this.page.locator('.list-unstyled').getByText(actionText, {exact: true});
+    await actionLocator.waitFor({state: 'visible', timeout: 5000});
+    await actionLocator.click();
```

Comment:
Combined with the page action stuff, this becomes
```ts
    const actionMenu = this.page.getByTestId('actionmenu');
    const actionLocator = actionMenu.getByText(actionText, {exact: true});

    await actionLocator.click();
```
As before, `.click()` already waits for the element to become visible.